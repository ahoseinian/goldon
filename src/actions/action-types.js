export const GET_ALL_ITEMS = 'GET_ALL_ITEMS'
export const GET_ALL_ITEMS_FULFILLED = 'GET_ALL_ITEMS_FULFILLED'
export const SET_SEARCH_VALUE = 'SET_SEARCH_VALUE'
export const RESET_SEARCH = 'RESET_SEARCH'
export const ADD_ITEM_TO_CART = 'ADD_ITEM_TO_CART'
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART'
export const TOGGLE_CART_VISIBILITY = 'TOGGLE_CART_VISIBILITY'
export const COMPLETE_CART_ORDER = 'COMPLETE_CART_ORDER'
