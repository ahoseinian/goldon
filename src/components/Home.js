import React from 'react';
import {connect} from 'react-redux'
import {Element} from 'react-scroll'
import {Helmet} from 'react-helmet'
import {getSearchedItems} from '../reducers/search'
import ItemList from './items/list'
import PageHead from './PageHead'

export const Home = ({items}) => (
  <div>
    <Helmet>
      <title>میاریم</title>
    </Helmet>
    <PageHead/>
    <div className="container">
      <Element name="firstInsideContainer">
        <ItemList items={items}/>
      </Element>
    </div>
  </div>
)

Home.propTypes = {
  items: React.PropTypes.arrayOf(
    React.PropTypes.object
  ).isRequired
}


export default connect((state) => ({
  items: getSearchedItems(state.items, state.search)
}))(Home)
