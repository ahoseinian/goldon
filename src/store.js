import {createStore, applyMiddleware} from 'redux'
import goldonApp from './reducers'
// import createLogger from 'redux-logger'
import createPromiseMiddleware from 'redux-promise-middleware'

// const logger = createLogger({collapsed: true});
const promiseMiddleware = createPromiseMiddleware();
const store = createStore(goldonApp, applyMiddleware(promiseMiddleware))


export default store
