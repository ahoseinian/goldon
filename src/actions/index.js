import request from 'superagent'
import {GET_ALL_ITEMS, SET_SEARCH_VALUE, ADD_ITEM_TO_CART, TOGGLE_CART_VISIBILITY, REMOVE_FROM_CART} from './action-types'

export const getAllItems = () => ({
  type: GET_ALL_ITEMS,
  payload: request.get('/api')
})

export const setSearchValue = (field, payload) => ({type: SET_SEARCH_VALUE, field, payload})
export const addItemToCart = (item) => ({type: ADD_ITEM_TO_CART, item})
export const toggleCartVisibility = () => ({type: TOGGLE_CART_VISIBILITY})
export const removeFromCart = (item) => ({type: REMOVE_FROM_CART, item})
