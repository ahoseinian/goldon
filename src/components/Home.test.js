import React from 'react'
import {shallow, mount} from 'enzyme'
import Home from './Home'
import store from '../store'
import {GET_ALL_ITEMS_FULFILLED} from '../actions/action-types'
import toJson from 'enzyme-to-json'

describe('Home', () => {
  it('renders', () => {
    const wrapper = shallow(<Home store={store}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders items', () => {
    store.dispatch({
      type: GET_ALL_ITEMS_FULFILLED,
      payload: {body:[
        {id:1},
        {id:2},
      ]}
    })

    const wrapper = shallow(<Home store={store}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
