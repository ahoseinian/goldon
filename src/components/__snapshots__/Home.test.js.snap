exports[`Home renders 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}>
    <title>
      میاریم
    </title>
  </HelmetWrapper>
  <PageHead />
  <div
    className="container">
    <Component
      name="firstInsideContainer">
      <List
        items={Array []} />
    </Component>
  </div>
</div>
`;

exports[`Home renders items 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}>
    <title>
      میاریم
    </title>
  </HelmetWrapper>
  <PageHead />
  <div
    className="container">
    <Component
      name="firstInsideContainer">
      <List
        items={
          Array [
            Object {
              "id": 1,
              "images": Array [],
            },
            Object {
              "id": 2,
              "images": Array [],
            },
          ]
        } />
    </Component>
  </div>
</div>
`;
